### ISSUE TITLE
Fix flake8 warning F841: local variable ... is assigned to but never used

### ISSUE THREAD

**Issue by vstinner** [python/cpython#76336]
BPO | [32155](https://bugs.python.org/issue32155)
--- | :---
Nosy | @vstinner, @merwok, @berkerpeksag, @serhiy-storchaka, @dstufft
PRs | <li>python/cpython#4608</li><li>python/cpython#4610</li><li>python/cpython#4618</li><li>python/cpython#4619</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-11-28.22:23:18.024>
created_at = <Date 2017-11-28.14:03:17.615>
labels = ['3.7', 'library']
title = 'Fix flake8 warning F841: local variable ... is assigned to but never used'
updated_at = <Date 2017-11-28.22:23:18.022>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-11-28.22:23:18.022>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-11-28.22:23:18.024>
closer = 'vstinner'
components = ['Library (Lib)']
creation = <Date 2017-11-28.14:03:17.615>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 32155
keywords = ['patch']
message_count = 10.0
messages = ['307120', '307123', '307124', '307125', '307128', '307144', '307168', '307177', '307179', '307180']
nosy_count = 5.0
nosy_names = ['vstinner', 'eric.araujo', 'berker.peksag', 'serhiy.storchaka', 'dstufft']
pr_nums = ['4608', '4610', '4618', '4619']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue32155'
versions = ['Python 2.7', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
Using flake8, I found some bugs in the standard libraries and tools.

I already fixed another flake8 error:
\---
commit 28e61650b23119b68cd7943ccc01b8b9af1b4103
Author: Victor Stinner \<victor.stinner@gmail.com\>
Date:   Tue Nov 28 00:34:08 2017 +0100

    bpo-31245: asyncio: Fix typo, isistance => isinstance (bpo-4594)
\---

Attached patch fix 3 bugs found by the flake8 F841 warning.

Note: Python 2.7 is not impacted by the turtledemo bug, but it's impacted by the two other bugs.

**Comment by vstinner** [comment]
New changeset 696b501cd11dc429a0f661adeb598bfaf89e4112 by Victor Stinner in branch 'master':
bpo-32155: Bugfixes found by flake8 F841 warnings (bpo-4608)
https://github.com/python/cpython/commit/696b501cd11dc429a0f661adeb598bfaf89e4112

**Comment by serhiy-storchaka** [comment]
If you have replaced self.DEFAULT_REALM with realm why not replace self.DEFAULT_REPOSITORY with repository?

I don't know what is correct. Added distutils experts for review.

**Comment by vstinner** [comment]
> If you have replaced self.DEFAULT_REALM with realm why not replace self.DEFAULT_REPOSITORY with repository?

Oh, I didn't notice that there is also a repository variable which is not used in all code paths.

> I don't know what is correct. Added distutils experts for review.

Now I'm not sure that my realm change is correct.

**Comment by berkerpeksag** [comment]
I don't think distutils changes are correct.

    ('realm', self.DEFAULT_REALM)

'realm' is the config name and 'self.DEFAULT_REALM' is its default value. In the 'for key, default in ...:' loop,

1. It checks if it's already set in the config file
2. Sets its value to 'current' if it does
3. If it doesn't, uses self.DEFAULT_REALM

distutils sets default values for undefined options via 'finalize_options' methods and there is the following command in 'PyPIRCCommand.finalize_options()' method:

```py
    if self.realm is None:
        self.realm = self.DEFAULT_REALM
```


So unless there is a user set value for the 'realm' option (see the 'upload' and 'register' commands), 'self.realm' equals to 'self.DEFAULT_REALM'.

I'd remove the following code instead:

```py
    realm = self.realm or self.DEFAULT_REALM
```

**Comment by merwok** [comment]
The best course of action would be to revert the distutils changes.

There are a handful of tickets related to register/upload commands, which are funky commands with their own config file not 100% in line with other distutils commands.  Changes in that code need tests.

**Comment by vstinner** [comment]
I proposed [PR 4618](https://github.com/python/cpython/pull/4618) to revert the distutils.config changes.

**Comment by vstinner** [comment]
New changeset 6979fcdc91114b1ccb16345e26734d6df4cccbc3 by Victor Stinner in branch '3.6':
bpo-32155: Bugfixes found by flake8 F841 warnings (bpo-4619)
https://github.com/python/cpython/commit/6979fcdc91114b1ccb16345e26734d6df4cccbc3

**Comment by vstinner** [comment]
New changeset 71bd588646b282c9eebc390b31184a5bdb54e712 by Victor Stinner in branch 'master':
bpo-32155: Revert distutils.config change (bpo-4618)
https://github.com/python/cpython/commit/71bd588646b282c9eebc390b31184a5bdb54e712

**Comment by vstinner** [comment]
The main bugs have been fixed in Python 3.6 and 3.7 (master).

Python 2.7 only has the bug in [Tools/scripts/treesync.py](https://github.com/python/cpython/blob/main/Tools/scripts/treesync.py), but I don't think that it's worth it to fix this old script in the stable 2.7 branch. I just removed this script from the master branch (bpo-32159).
